util/spot.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"net/http"
+
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+// SpotHostWillTerminateSoon returns true if the EC2 spot host it is running on will terminate soon.
+func SpotHostWillTerminateSoon() bool {
+	const url = "http://169.254.169.254/latest/meta-data/spot/termination-time"
+	c := GetHTTPClient()
+	defer PutHTTPClient(c)
+	resp, err := c.Get(url)
+	if err != nil {
+		grip.Info(errors.Wrap(err, "problem getting termination endpoint"))
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusOK {
```

Comment:
can we grip log the content of the body? I suppose it doesn't matter

Reply:
I can't imagine we'd be able to get to it in time. The host is about to shut down, and there's no way to stop it.